BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/rootAnalysis.py

```diff
@@ -1,297 +1,238 @@
-
-	with open(g.opath+"/"+str(g.gen)+"_pueoOut.csv",'a') as f:
-		f.write(str(g.indiv)+","+str(effective_V[0])+","+str(effective_V_p[0])+","+str(effective_V_m[0])+"\n")
-		print("Writing to:", g.opath+"/"+str(g.gen)+"_pueoOut.csv" )
-
-	with open(g.opath+"/"+str(g.gen)+"_fitnessScores.csv",'a') as f:
-		f.write(str(effective_V[0])+"\n")
-		print("Writing to:", g.opath+"/"+str(g.gen)+"_fitnessScores.csv")
-
-	with open(g.opath+"/"+str(g.gen)+"_vEffectives.csv",'a') as f:
-		f.write(str(effective_V[0])+"\n")
-		print("Writing to:", g.opath+"/"+str(g.gen)+"_fitnessScores.csv")
-
-	with open(g.opath+"/"+str(g.gen)+"_errorBars.csv",'a') as f:
-		f.write(str(effective_V_p[0])+","+str(effective_V_m[0])+"\n")
-		print("Writing to:", g.opath+"/"+str(g.gen)+"_errorBars.csv")
-
-	'''
-	effective_A=(effective_V/(L_int/100000))#factor of 100000 to convert from cm to km 
-	#print(effective_A)
-	plt.figure(1)
-	plt.plot(E_EV,effective_V,'o-',color='black')
-	plt.grid()
-	plt.xlabel('Log( Energy [eV] )')
-	plt.ylabel('Effective Volume [$km^3 sr$]')
-	plt.yscale('log')
-	plt.savefig('effectiveV.pdf')
-
-	#interaction_lengths=L_int/1e-3
-
-	#print((effective_V-effective_V_m)/interaction_lengths*1e3,(effective_V_p-effective_V)/interaction_lengths*1e3)
-	plt.figure(2)
-	#plt.plot(E_EV,effective_V/interaction_lengths*1e3,'o',color='black',label='PUEOSim $A_{eff}$')
-	#plt.plot(energies,effective_A,color='dodgerblue')
-	plt.ylabel('Effective Area [$km^2 sr$]')
-	plt.xlabel('Log( Energy [eV] )')
-	#expected_Aeff = np.loadtxt('diffuse_Askaryan.txt')
-	#plt.plot(np.log10(expected_Aeff[:,0]),expected_Aeff[:,1],color='dodgerblue',label="Scaled $A_{eff}$")
-	plt.errorbar(E_EV,effective_V/interaction_lengths*1e3,yerr=[effective_V_m/interaction_lengths*1e3,effective_V_p/interaction_lengths*1e3],color='black',label='PUEOSim $A_{eff}$')
-	
-	np.savez('effectiveArea.npz', E_EV, effective_V/interaction_lengths*1e3, [effective_V_m/interaction_lengths*1e3,effective_V_p/interaction_lengths*1e3])
-
-	plt.yscale('log')
-	plt.grid()
-	plt.legend()
-	plt.xlim(17.5, 21.0)
-	plt.ylim(1e-5, 1e3)
-	plt.savefig('effectiveA'+thisLabel+'.pdf')
-	#plt.close()
-
-	#plt.show()
-	plt.figure(3)
-	plt.plot(E_EV,effective_V/interaction_lengths*1e3,'o-',color=thisColor,label=thisLabel)
-	#print("effective As", effective_V/interaction_lengths*1e3)
-	#plt.errorbar(energies,effective_V/interaction_lengths*1e3,yerr=[(effective_V-effective_V_m)/interaction_lengths*1e3,(effective_V_p-effective_V)/interaction_lengths*1e3],color=thisColor,label=thisLabel)
-
-	np.save('effectiveA_'+str(thisLabel)+'.npy',effective_V/interaction_lengths*1e3)
-	np.save('numPass_'+str(thisLabel)+'.npy',all_passed)
-	np.save('numWeight_'+str(thisLabel)+'.npy',all_weighted)
-	np.save('totalEvents_'+str(thisLabel)+'.npy',all_events)
-	np.save('squaredWeights_'+str(thisLabel)+'.npy',all_weights_squared)
-	'''
+    root = g.WorkingDir + "/Run_Outputs/" + g.RunName + "/Root_Files/" + str(g.gen) + "_Root_Files"
```

Comment:
If you use `type=Path` for the `workingdir` above (btw, it shouldn't be CamelCase), then you can do 
```suggestion
    root = g.WorkingDir / "Run_Outputs" / g.RunName / "Root_Files" / f"{g.gen}_Root_Files"
```

Reply:
I'm not able to use f strings, but I will replace them to use .format()